File: Map/Map/Map/view/TextMenu.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Map.view
{
    class TextMenu
    {
        private Dictionary<String, Command> commands;

        public TextMenu()
        {
            commands = new Dictionary<string, Command>();
        }

        public void addCommand(Command c)
        {
            commands.Add(c.key, c);
        }

        private void printMenu()
        {
            foreach(Command c in commands.Values)
            {
                Console.WriteLine("{0} : {1}", c.key, c.description);
            }
        }

        public void show()
        {
            while(true)
            {
                printMenu();
                Console.Write("Enter option: ");
                String key = Console.ReadLine();
                Command c = commands[key];
                if (c == null)
                    Console.WriteLine("Invalid option!");
                else
                    c.execute();
            }
        }
    }
}

File: Map/Z_InterpretorWCDL/src/view/TextMenu.java
package view;

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;


public class TextMenu 
{
	private Map<String,Command> commands;
	
	public TextMenu()
	{
		commands = new HashMap<String,Command>();
	}
	
	public void addCommand(Command c)
	{
		commands.put(c.getKey(),c);
	}
	
	private void printMenu()
	{
		for(Command com : commands.values())
		{
			String line = String.format("%4s : %s",com.getKey(),com.getDescription());
			System.out.println(line);
		}
	}
	
	public void show()
	{
		try(Scanner sc = new Scanner(System.in);)
		{
		while(true)
		{
			printMenu();
			System.out.println("Enter your option: ");
			String key = sc.nextLine();
			Command com = commands.get(key);
			if(com==null)
			{
				System.out.println("Invalid option.\n");
			}
			else
				com.execute();
		}
		}
		
		
	}
}
